python/adjoint/filters.py

```diff
@@ -112,21 +131,15 @@ def cylindrical_filter(x, radius, Lx, Ly, resolution):
     [1] Lazarov, B. S., Wang, F., & Sigmund, O. (2016). Length scale and manufacturability in
     density-based topology optimization. Archive of Applied Mechanics, 86(1-2), 189-218.
     """
-    Nx = int(Lx * resolution)
-    Ny = int(Ly * resolution)
+    Nx = int(round(Lx * resolution))
+    Ny = int(round(Ly * resolution))
     x = x.reshape(Nx, Ny)  # Ensure the input is 2D
 
     xv = np.arange(0, Lx / 2, 1 / resolution)
     yv = np.arange(0, Ly / 2, 1 / resolution)
 
-    cylindrical = lambda a: np.where(a <= radius, 1, 0)
-    hx = cylindrical(xv)
-    hy = cylindrical(yv)
-
-    h = np.outer(_proper_pad(hx, 3 * Nx), _proper_pad(hy, 3 * Ny))
```

Comment:
I think we should still be able to call "1d" `_proper_pad` functions along each direction?

That is, when you pad along `x`, you don't need to "know" about the `y` direction — you just need the diameter of the kernel in `x`.  And vice versa for `y`.